src/coreclr/tools/Common/Microsoft/SourceLink/Tools/SourceLinkMap.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.SourceLink.Tools
+{
+    /// <summary>
+    /// Source Link URL map. Maps file paths matching Source Link patterns to URLs.
+    /// </summary>
+    internal readonly struct SourceLinkMap
```

Comment:
This file is basically https://github.com/dotnet/sourcelink/blob/928f459936292b3128b949957c7b46ae5ed59e4c/src/SourceLink.Tools/SourceLinkMap.cs, but fixed up not to trigger warnings.

We can't use this in the NuGet form because for whatever reason this is a source code nuget (doesn't have an assembly, it contains C# source code) that triggers tons of warnings in our build. So a fork it is.

Reply:
It would be preferable if it was not a source code nuget. This will be an ongoing problem in source form.